In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

df.head()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [3]:
df.shape

(100, 13)

In [4]:
df.columns

Index(['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category',
       'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales',
       'Rating', 'Payment_Method'],
      dtype='object')

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    object 
 2   Customer          100 non-null    object 
 3   City              100 non-null    object 
 4   Region            100 non-null    object 
 5   Category          100 non-null    object 
 6   Product           100 non-null    object 
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12  Payment_Method    100 non-null    object 
dtypes: float64(2), int64(4), object(7)
memory usage: 10.3+ KB


In [6]:
df.describe()

,Order_ID,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating
count,100.000000,100.000000,100.000000,100.000000,100.000000,100.000000
mean,1050.500000,2.940000,1647.000000,6.500000,4472.629500,4.322000
std,29.011492,1.391315,970.054669,6.092718,3324.306691,0.414285
min,1001.000000,1.000000,499.000000,0.000000,449.100000,3.500000
25%,1025.750000,2.000000,799.000000,0.000000,1948.050000,4.100000
50%,1050.500000,3.000000,1499.000000,5.000000,3228.300000,4.300000
75%,1075.250000,4.000000,2499.000000,11.250000,6427.562500,4.700000
max,1100.000000,5.000000,3499.000000,15.000000,13995.000000,5.000000


In [7]:
df["Product"]

,Product
0,Wireless Headphones
1,Running Shoes
2,Power Bank
3,Mechanical Keyboard
4,Smart Watch
...,...
95,Dumbbell Set
96,T-Shirt
97,Electric Kettle
98,Machine Learning Basics


In [9]:
df[["Product", "Category", "Total_Sales"]]

,Product,Category,Total_Sales
0,Wireless Headphones,Electronics,4497.00
1,Running Shoes,Sports,2799.00
2,Power Bank,Electronics,2038.30
3,Mechanical Keyboard,Electronics,9496.20
4,Smart Watch,Electronics,3134.05
...,...,...,...
95,Dumbbell Set,Sports,9495.25
96,T-Shirt,Clothing,679.15
97,Electric Kettle,Home & Kitchen,5996.00
98,Machine Learning Basics,Books,1998.00


In [10]:
df[df["Total_Sales"] > 1000]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
...,...,...,...,...,...,...,...,...,...,...,...,...,...
94,1095,2026-02-25,Maryam Khan,Mumbai,West,Electronics,Power Bank,2,1199,0,2398.00,4.6,Debit Card
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI


In [11]:
df[df["Category"] == "Electronics"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI
25,1026,2026-05-12,Sara Ahmed,Chandigarh,North,Electronics,Wireless Headphones,3,1499,0,4497.00,4.2,Net Banking
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI


In [13]:
df.sort_values("Total_Sales", ascending=False)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
...,...,...,...,...,...,...,...,...,...,...,...,...,...
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
73,1074,2026-03-28,Alina Shah,Lucknow,North,Books,Atomic Habits,1,599,10,539.10,5.0,Net Banking
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery


In [14]:
df.sort_values("Total_Sales", ascending=False).head(10)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
47,1048,2026-06-24,Sara Ahmed,Lucknow,North,Home & Kitchen,Coffee Maker,3,3499,0,10497.00,5.0,UPI
69,1070,2026-06-30,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,3.8,Net Banking
79,1080,2026-05-25,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,4.6,Debit Card
51,1052,2026-03-13,Rohan Mehta,Chandigarh,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.2,Debit Card
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card


In [16]:
category_sales = df.groupby("Category")["Total_Sales"].sum()

category_sales.sort_values(ascending=False)

,Total_Sales
Category,
Sports,136026.45
Home & Kitchen,113329.90
Electronics,92086.00
Clothing,69028.20
Books,36792.40


In [17]:
city_sales = df.groupby("City")["Total_Sales"].sum()

city_sales.sort_values(ascending=False)

,Total_Sales
City,
Hyderabad,59076.45
Chennai,47493.20
Lucknow,46194.65
Pune,41908.55
Mumbai,41457.80
Kochi,37257.50
Chandigarh,34539.65
Delhi,30882.80
Jaipur,25534.00


In [18]:
product_sales = df.groupby("Product")["Total_Sales"].sum()

product_sales.sort_values(ascending=False)

,Total_Sales
Product,
Coffee Maker,51610.25
Jacket,47106.15
Cricket Bat,41483.40
Running Shoes,41285.25
Wireless Headphones,33877.40
Dumbbell Set,30184.90
Mechanical Keyboard,26489.40
Electric Kettle,26382.40
Mixer Grinder,25341.55


In [19]:
payment_sales = df.groupby("Payment_Method")["Total_Sales"].sum()

payment_sales.sort_values(ascending=False)

,Total_Sales
Payment_Method,
Credit Card,106540.25
Debit Card,98064.90
Net Banking,96371.25
UPI,96163.25
Cash on Delivery,50123.30


In [20]:
total_sales = df["Total_Sales"].sum()

print("Total Sales:", total_sales)

Total Sales: 447262.95000000007


In [21]:
average_sales = df["Total_Sales"].mean()

print("Average Sales:", average_sales)

Average Sales: 4472.629500000001


In [23]:
print("Highest Sale:", df["Total_Sales"].max())
print("Lowest Sale:", df["Total_Sales"].min())

Highest Sale: 13995.0
Lowest Sale: 449.1


In [24]:
total_quantity = df["Quantity"].sum()

print("Total Quantity Sold:", total_quantity)

Total Quantity Sold: 294


In [25]:
number_of_orders = len(df)

print("Number of Orders:", number_of_orders)

Number of Orders: 100


In [27]:
df.groupby("Category")["Total_Sales"].agg(
    ["sum", "mean", "max", "min"]
)

,sum,mean,max,min
Category,,,,
Books,36792.40,1752.019048,4495.00,449.10
Clothing,69028.20,4930.585714,10620.75,679.15
Electronics,92086.00,4846.631579,9496.20,1708.10
Home & Kitchen,113329.90,5396.661905,12596.40,1188.30
Sports,136026.45,5441.058000,13995.00,759.05


In [29]:
top_category = (
    df.groupby("Category")["Total_Sales"]
    .sum()
    .sort_values(ascending=False)
    .head(1)
)

print(top_category)

Category
Sports    136026.45
Name: Total_Sales, dtype: float64


In [30]:
top_product = (
    df.groupby("Product")["Total_Sales"]
    .sum()
    .sort_values(ascending=False)
    .head(1)
)

print(top_product)

Product
Coffee Maker    51610.25
Name: Total_Sales, dtype: float64


The dataset contains 100 orders and generates total sales of ₹447,262.95.
The average sales value per order is approximately ₹4,472.63.
Sports is the highest-performing category with total sales of ₹136,026.45.
Hyderabad has the highest total sales among all cities, generating ₹59,076.45.
Coffee Maker is the top-performing product with sales of ₹51,610.25.
Credit Card is the payment method associated with the highest sales, contributing ₹106,540.25.
A total of 294 units were sold across the 100 orders.
The highest individual order value is ₹13,995, while the lowest is ₹449.10.
The dataset contains no missing values, so the data is complete for the requested analysis.
Sports, Home & Kitchen, and Electronics are the three largest categories by total sales.